airbyte-webapp/src/services/job/JobService.tsx

```diff
@@ -39,11 +39,25 @@ export const useGetJob = (id: number, enabled = true) => {
   });
 };
 
-export const useGetDebugInfoJob = (id: number, enabled = true): JobDebugInfoRead | undefined => {
+export const useGetDebugInfoJob = (
+  id: number,
+  enabled = true,
+  refetchWhileRunning = false
+): JobDebugInfoRead | undefined => {
   const service = useGetJobService();
 
   return useSuspenseQuery(jobsKeys.getDebugInfo(id), () => service.getDebugInfo(id), {
-    refetchInterval: false,
+    refetchInterval: !refetchWhileRunning
+      ? false
+      : (data) => {
+          // If refetchWhileRunning was true, we keep refetching debug info (including logs), while the job is still
+          // running or hasn't ended too long ago. We need some time after the last attempt has stopped, since logs
+          // keep incoming for some time after the job has already been marked as finished.
+          const lastAttemptEndTimestamp = data?.attempts[data.attempts.length - 1].attempt.endedAt;
+          // While no attempt ended timestamp exists yet (i.e. the job is still running) or it hasn't ended
+          // more than 2 minutes (2 * 60 * 1000ms) ago, keep refetching
+          return lastAttemptEndTimestamp && Date.now() - lastAttemptEndTimestamp * 1000 > 2 * 60 * 1000 ? false : 2500;
```

Comment:
The PR description says 6 minutes, but this looks like 2.  Is one of them incorrect?

Reply:
Sorry, forgot to update that :) I created this originally with 6 minutes, but Davin clarified for me, that the log is supposed to be max 1 minute late based on how our log fetching implementation worked, so I felt 2 minutes might be a long enough grace period to make sure.